nemo/collections/tts/data/vocoder_dataset.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import traceback
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
+
+import librosa
+import torch.utils.data
+
+from nemo.collections.asr.parts.preprocessing.segment import AudioSegment
+from nemo.collections.asr.parts.utils.manifest_utils import read_manifest
+from nemo.collections.tts.parts.preprocessing.feature_processors import FeatureProcessor
+from nemo.collections.tts.parts.utils.tts_dataset_utils import (
+    filter_dataset_by_duration,
+    get_abs_rel_paths,
+    get_weighted_sampler,
+    stack_tensors,
+)
+from nemo.core.classes import Dataset
+from nemo.utils import logging
+from nemo.utils.decorators import experimental
+
+
+@dataclass
+class DatasetMeta:
+    manifest_path: Path
+    audio_dir: Path
+    sample_weight: float = 1.0
+
+
+@dataclass
+class DatasetSample:
+    manifest_entry: dict
+    audio_dir: Path
+
+
+@experimental
+class VocoderDataset(Dataset):
+    def __init__(
+        self,
+        dataset_meta: Dict[str, DatasetMeta],
+        sample_rate: int,
+        n_segments: Optional[int] = None,
+        weighted_sample_steps: Optional[int] = None,
+        feature_processors: Optional[Dict[str, FeatureProcessor]] = None,
+        min_duration: Optional[float] = None,
+        max_duration: Optional[float] = None,
+        num_audio_retries: int = 5,
+    ):
+        super().__init__()
+
+        self.sample_rate = sample_rate
+        self.n_segments = n_segments
+        self.weighted_sample_steps = weighted_sample_steps
+        self.num_audio_retries = num_audio_retries
+        self.load_precomputed_mel = False
+
+        if feature_processors:
+            logging.info(f"Found feature processors {feature_processors.keys()}")
+            self.feature_processors = feature_processors.values()
+        else:
+            self.feature_processors = []
+
+        self.data_samples = []
+        self.sample_weights = []
+        for dataset_name, dataset in dataset_meta.items():
+            samples, weights = self._process_dataset(
+                dataset_name=dataset_name, dataset=dataset, min_duration=min_duration, max_duration=max_duration,
+            )
+            self.data_samples += samples
+            self.sample_weights += weights
+
+    def get_sampler(self, batch_size: int) -> Optional[torch.utils.data.Sampler]:
```

Comment:
Also looks like a `@property`

Reply:
For it to be a property it cannot take `batch_size` as input. Where the batch size (and sampler) are inputs to the pytorch DataLoader rather than the Dataset.

The main alternative would be to move this function out out of the Dataset class and into `dataset_utils`.

Reply:
It is called in the model/ classes, so It can be reused in multiple training different training recipes. Right now in this PR FastPitch is the only one using the TTS sampler and HiFi-GAN the vocoder, but I will be adding it to UnivNet and EnCodec recipes soon and if we deprecate old data loader eventually then all recipes would use it.

Reply:
OK, if you think having this method is preferable to directly calling `get_weighted_sampler`, let's leave it as is